cedar-policy-validator/src/human_schema/to_json_schema.rs

```diff
@@ -488,31 +478,28 @@ fn search_cedar_namespace(name: SmolStr, loc: Loc) -> Result<SchemaType, ToJsonS
             Ok(SchemaType::Type(SchemaTypeVariant::Extension { name }))
         }
         _ => Err(ToJsonSchemaError::UnknownTypeName(Node::with_source_loc(
-            name, loc,
+            name.to_smolstr(),
+            loc,
         ))),
     }
 }
 
 #[derive(Default)]
 struct NamespaceRecord {
-    entities: HashMap<SmolStr, Node<()>>,
-    common_types: HashMap<SmolStr, Node<()>>,
+    entities: HashMap<Id, Node<()>>,
+    common_types: HashMap<Id, Node<()>>,
     loc: Option<Loc>,
```

Comment:
Is this (and similar changes) a breaking change?  `SmolStr` would allow all strings, while `Id` disallows a number of special characters, spaces, and certain keywords like `true`, `in`, and `has`.  I'm most interested about the keywords actually.  Were they previously valid namespaces in the human schema format, and now are not?

Reply:
@john-h-kastner-aws  can confirm this but my understanding is that they need to these restrictions are needed.